src/subdomains/generic/kyc/dto/mapper/kyc-info.mapper.ts

```diff
@@ -14,25 +15,30 @@ import { KycLevelDto, KycSessionDto } from '../output/kyc-info.dto';
 import { KycStepMapper } from './kyc-step.mapper';
 
 export class KycInfoMapper {
-  static toDto(userData: UserData, withSession: false, currentStep?: KycStep): KycLevelDto;
-  static toDto(userData: UserData, withSession: true, currentStep?: KycStep): KycSessionDto;
-
-  static toDto(userData: UserData, withSession: boolean, currentStep?: KycStep): KycLevelDto | KycSessionDto {
+  static toDto(
+    userData: UserData,
+    withSession: boolean,
+    kycClients: Wallet[],
+    currentStep?: KycStep,
+  ): KycLevelDto | KycSessionDto {
     const kycSteps = KycInfoMapper.getUiSteps(userData);
     currentStep ??=
       kycSteps.find((s) => s.status === KycStepStatus.IN_PROGRESS) ??
       kycSteps.find((s) => s.status === KycStepStatus.FAILED);
 
+    const userKycClients = kycClients?.filter((kc) => userData.kycClientList.includes(kc.id));
+
```

Comment:
`userData.kycClientList` can be undefined. Maybe we should add a fallback in the getter, would be safer.